tests/unit_tests/test_deplete_batchwise.py

```diff
@@ -0,0 +1,204 @@
+import numpy as np
+
+import openmc
+import openmc.lib
+from openmc.deplete import CoupledOperator
+from openmc.deplete import (BatchwiseCellGeometrical, BatchwiseCellTemperature,
+    BatchwiseMaterialRefuel)
+
+CHAIN_PATH = Path(__file__).parents[1] / "chain_simple.xml"
+
+@pytest.fixture
+def model():
+    f = openmc.Material(name="fuel")
+    f.add_element("U", 1, percent_type="ao", enrichment=4.25)
+    f.add_element("O", 2)
+    f.set_density("g/cc", 10.4)
+    f.temperature = 293.15
+
+    w = openmc.Material(name="water")
+    w.add_element("O", 1)
+    w.add_element("H", 2)
+    w.set_density("g/cc", 1.0)
+    w.temperature = 273.15
+    w.depletable = True
+
+    h = openmc.Material(name='helium')
+    h.add_element('He', 1)
+    h.set_density('g/cm3', 0.001598)
+
+    radii = [0.42, 0.45]
+    height = 0.5
+
+    f.volume = np.pi * radii[0] ** 2 * height
+    w.volume = np.pi * (radii[1]**2 - radii[0]**2) * height/2
+
+    materials = openmc.Materials([f, w, h])
+
+    surf_interface = openmc.ZPlane(z0=0)
+    surf_top = openmc.ZPlane(z0=height/2)
+    surf_bot = openmc.ZPlane(z0=-height/2)
+    surf_in = openmc.Sphere(r=radii[0])
+    surf_out = openmc.Sphere(r=radii[1], boundary_type='vacuum')
+
+    cell_water = openmc.Cell(fill=w, region=-surf_interface)
+    cell_helium = openmc.Cell(fill=h, region=+surf_interface)
+    universe = openmc.Universe(cells=(cell_water, cell_helium))
+    cell_fuel = openmc.Cell(name='fuel_cell', fill=f,
+                            region=-surf_in & -surf_top & +surf_bot)
+    cell_universe = openmc.Cell(name='universe_cell',fill=universe,
+                            region=+surf_in & -surf_out & -surf_top & +surf_bot)
+    geometry = openmc.Geometry([cell_fuel, cell_universe])
+
+    settings = openmc.Settings()
+    settings.particles = 1000
+    settings.inactive = 10
+    settings.batches = 50
+
+    return openmc.Model(geometry, materials, settings)
+
+@pytest.mark.parametrize("case_name, obj, attribute, bracket, limit, axis, vec", [
+    ('cell translation','universe_cell', 'translation', [-1,1], [-10,10], 2, None),
+    ('cell rotation', 'universe_cell', 'rotation', [-1,1], [-10,10], 2, None),
+    ('single cell temperature', 'fuel_cell', 'temperature', [-1,1], [-10,10], 2, None ),
+    ('multi cell', 'universe_cell', 'temperature',  [-1,1], [-10,10], 2, None ),
+    ('material refuel', 'fuel', 'refuel', [-1,1], [-10,10], None, {'U235':0.1, 'U238':0.9}),
+    ('invalid_1', 'universe_cell', 'refuel', [-1,1], [-10,10], None, {'U235':0.1, 'U238':0.9}),
+    ('invalid_2', 'fuel', 'temperature',  [-1,1], [-10,10], 2, None ),
+    ('invalid_3', 'fuel', 'translation', [-1,1], [-10,10], 2, None),
+    ])
+def test_attributes(case_name, model, obj, attribute, bracket, limit, axis, vec):
+    """
+    Test classes attributes are set correctly
+    """
+    op = CoupledOperator(model, CHAIN_PATH)
+
+    int = openmc.deplete.PredictorIntegrator(
+        op, [1,1], 0.0, timestep_units = 'd')
```

Comment:
This integrator is used a lot in this file. Could we make it a `pytest.fixture` instead?